Orange/widgets/visualize/owscatterplotgraph.py

```diff
@@ -1052,13 +1049,31 @@ def update_labels(self):
             ti.setPos(xp, yp)
             self.plot_widget.addItem(ti)
             self.labels.append(ti)
-        self._signal_too_many_labels(False)
 
     def _signal_too_many_labels(self, too_many):
         if self._too_many_labels != too_many:
             self._too_many_labels = too_many
             self.too_many_labels.emit(too_many)
 
+    def _label_mask(self, x, y):
+        (x0, x1), (y0, y1) = self.view_box.viewRange()
+        mask = np.logical_and(
+            np.logical_and(x >= x0, x <= x1),
+            np.logical_and(y >= y0, y <= y1))
+        if self.label_only_selected:
+            sub_mask = self._filter_visible(self.master.get_subset_mask())
```

Comment:
Aaaa, having masks that can be None make the code so ugly... But I did not find a better solution. 

Perhaps in the future we could have some helper functions for logical_* where arguments could be None? Or just have masks that are never None?